src/nodes.coffee

```diff
@@ -23,6 +23,10 @@ NO      = -> no
 THIS    = -> this
 NEGATE  = -> @negated = not @negated; this
 
+# Track comments that have been compiled into fragments, to avoid outputting
+# them twice.
+compiledComments = []
```

Comment:
I’ve moved this variable to be a property of `Block` just in case. I tested with this code and got the same output both times:

```coffee
CoffeeScript = require './lib/coffeescript'

coffee = '''
  # 1
  return unless window?

  ### 2 ###
  return if global?
  '''

console.log CoffeeScript.compile coffee
console.log CoffeeScript.compile coffee
```